File: Problems/0009_Palindrome/Project_CS/Palindrome.cs
using System;

public class Solution {
    public bool IsPalindrome(int x) {
        string temp = x.ToString();
        
        for (int i = 0; i < temp.Length / 2; i++) {
            if (temp[i] != temp[temp.Length - 1 - i])
                return false;
        }
        
        return true;
    }

    public void Main(string args)
    {
        int x = int.Parse(args);

        Console.WriteLine("x = " + x.ToString());

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();
        
        bool result = IsPalindrome(x);

        Console.WriteLine("result = " + result.ToString());
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}

File: Problems/0001_0099/0009_Palindrome/Project_Python3/Palindrome.py
import os
import sys
import time

class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        return str(x)[:] == str(x)[::-1]

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    x = int(temp.replace("\"","").replace("[","").replace("]","").rstrip())
    print("x = {0}".format(x))

    sl = Solution()
    time0 = time.time()

    result = sl.isPalindrome(x)

    time1 = time.time()

    print("result = {0}".format(result))
    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
